### DISCUSSION TITLE
Maven perspective / Maven EPP

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-m2e/m2e-core#1016] [Ideas]
Its often told that other IDEs servers users better, because they are simpler / less confusing / more focused / ...

Because of this, I wonder if it would be useful to have a Maven perspective (or even a Maven EPP), it does not have to be much different than the "standard" but probably can hide some of the options and make others more prominent.

**Comment by HannesWell** [comment]
A Maven perspective sounds good to me, a own EPP is IMHO a bit too much and would require a user to already choose the right product when installing Eclipse. A perspective would allow to change later and would also make that available to all EPPs that include M2E.

Besides highlighting and hiding some Import- and other Options, that perspective can also high-light some M2E specific views like
- `Maven Lifecycle Mapping`
- `Maven Repositories`
- `Maven Workspace Build`

If we add a perspective we should use E4.

**Comment by mickaelistria** [comment]
Other IDEs don't have a strong concept of perspectives or EPP. IMO multiplication of perspectives and products is exactly what makes the Eclipse IDE *less* focused on user needs but instead artificially exposing some internals (which Eclipse.org projects contributes this or that; what are the internal concepts of the plugin...). I think perspective are useful only if using some features efficiently cannot be achieved with the standard "Resources" perspective -and I don't think it's the case here-; and an EPP is IMO more than overkill as I don't get why a user will want to download a "Maven IDE" when they can already have a complete Java IDE.
More is often the opposite of simpler: giving user choices, perspecties, IDEs... does increase the amount of decision they have to take, ultimately increasing their confusion and reducing their efficiency. Other IDEs don't have a dedicated Maven perspective, they don't have a dedicated Maven IDE and yet people like to say they like them because they're more efficient. This is a better pattern to mimic; mulitplying perspectives and IDEs is how Eclipse worked 15 years ago, when it started loosing more than half its marketshare...

**Reply by laeubi** [reply]
> I think perspective are useful only if using some features efficiently cannot be achieved with the standard "Resources" perspective -and I don't think it's the case here-

- Just download eclipse, and open a new Workspace. The first thing you see is the welcome screen
- The first option (for RCP/RAP EPP I have no other edition at hand right now so just take it as an example) is a Tutorial creating an RCP Application (with maven?? most probably not)
- Second is RAP (maven anyone?)
- next is "New plugin project" (maven??)
- Then checkout from git .. maybe maven
- "Import existing Porject" ... that sounds promising, so I can try out my maven project right now!
- So it opens a dialog, i must choose browse (what other choice I actually have??) and I now select my cool maven project
- Eclipse tells you "no projects found" ... WTF?
- After cancel I might give it another try (I see now "create project", so if Eclipse is to stupid to open my maven project it should be easy to create a new one
- On the Top I now see "Empty Composer project", "PHP Project" (seems I installed PHP one time into my Eclipse)
- Oh... there is also a Maven folder so finally... MAVEN somewhere! Alright Checkout from SCM?? For a new Project? Anyways choose "New Maven project"
- WTF Java 1.5? Really?
- So how run maven build?
- Why is there so much Hidden under "Maven", Why some stuff under "Run/Debug" ... why hidden sub-menus ... maybe Eclipse is not made for maven?
- Well lets try another one so File > New > ... lots of stuff but no maven .. :-1: 

So I think there is a lot one could improve for maven user compared to the standard "resources" perspective that makes working with maven more clean and easy...

**Reply by mickaelistria** [reply]
All those are usability bugs to fix here and there, definitely not something a perspective or another .zip file would help at solving.
Focusing effort on fixing those usability issues would actually be extremely more profitable to all Eclipse IDE users than creating some forked workflows specifically for Maven.

**Reply by laeubi** [reply]
In theory, yes, in practice there are so many concerns to fix something "in general" that actually either nothing happens or it only happens after decades of discussions and there is always someone not happy why the change and complains.

**Reply by mickaelistria** [reply]
Adding perspectives and other attempts to workaround usability just make usability issues bigger and bigger by contributing new usability issues. We're now paying the debt of those who've kept stacking project-specific workarounds instead of improving Platform back then. We need to learn from history and not keep repeating mistakes; even if doing things right seems harder.
